### ISSUE TITLE
python-format flag incorrectly forced on all extractors

### ISSUE THREAD

**Issue by wichert** [python-babel/babel#35]
This is a copy of [old ticket 318](http://babel.edgewall.org/ticket/318).

During extraction the results from an extractor plugin are used to create a message which is then added to the catalog. When the Message instance is created it can set the `python-format` flag if the text matches a specific regular expression. That means that a text like this:

```
This is a text about 50% of all users.
```

will be flagged as a python-format string even it does not originate from python code, or from a piece of python code that will never use formatting. This is breaking our translations in a pretty bad way currently.

As far as I can see the extractor is the only thing that should decide the flags for a message. The Message class itself should never try to guess flags or force them upon plugins, especially since extractors have no way to override this behaviour.

**Comment by Changaco** [comment]
> As far as I can see the extractor is the only thing that should decide the flags for a message.

I disagree. For example, how would the python extractor know whether the flag should be `python-format` or `python-brace-format` (#333)? It can only guess, and not any better than the `Message` class can.

`xgettext` has a `--flag` argument that's supposed to handle this ([doc](https://www.gnu.org/software/gettext/manual/html_node/xgettext-Invocation.html)). Calling `xgettext -o - --flag=_:1:python-format file.py`, with `file.py` containing calls like `_("Some message")`, results in all those messages being flagged with `python-format`.

**Comment by lelit** [comment]
Is there a particular reason why the [PYTHON_FORMAT](https://github.com/python-babel/babel/blob/master/babel/messages/catalog.py#L33) regexp accepts whitespaces _after_ the initial `%` sign?
That's the main reason why wichert's example at the start (that is, `This is a text about 50% of all users.`) is being marked with `python-format`...
For now I replaced that with a `FULLWIDTH PERCENT SIGN`, to not trigger an error at compile time, but I wonder if the optional spaces should be removed from the regexp instead.

**Comment by wichert** [comment]
Because Python allows it:

```
Python 2.7.15 (default, Jun 17 2018, 12:46:58) 
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> '% d' % 1
' 1'
```

Note the leading whitespace in the result, which is significant here. This is inherited from C, which also allows it:

```C
#include <stdio.h>
#include <stdlib.h>

int main() {
	printf("% d\n", 1);
	return 0;
}
```

produces:

```
wichert@notos /tmp $ make x
cc     x.c   -o x
wichert@notos /tmp $ ./x
 1
```

**Comment by lelit** [comment]
Oh, didn't know about that! Thanks!
